core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -180,14 +180,24 @@ object Controller {
         // second argument.  (TODO .. seems fragile)
         val instance = if (args.length > 0) args(1).toInt else 0
 
-        // initialize the runtimes manifest
-        if (config.isValid && ExecManifest.initialize(config)) {
-            val port = config.servicePort.toInt
-            BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
-        } else {
+        def abort() = {
             logger.error(this, "Bad configuration, cannot start.")
             actorSystem.terminate()
             Await.result(actorSystem.whenTerminated, 30.seconds)
         }
+
+        if (!config.isValid) {
+            abort()
+            return
+        }
+
```

Comment:
you could rewrite the rest as 

```
ExecManifest.initialize(config) match {
            case Success(_) =>
                val port = config.servicePort.toInt
                BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))

            case Failure(t) =>
                logger.error(this, s"Invalid runtimes manifest: ${execManifest.failed.get}")
                abort()
        }
```

i can live with the `return`s since the scope is small and limited here.